### PR TITLE
Add synchronizing of VNet peering when sync level is `LocalNotInSync`

### LINKED ISSUE #1023: Automatically synchronize peered VNet using the azure.azcollection.azure_rm_virtualnetworkpeering module
##### SUMMARY
When adding a new CIDR block to an existing VNet that is already peered, provide us a way to resync the peering.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
azure_rm_virtualnetworkpeering

##### ADDITIONAL INFORMATION
It would be used to resync the network peering when we add more address space to a VNet.

```yaml

```

### PR DESCRIPTION
**Pull request by kmj251**
##### SUMMARY

Added a method to sync VNet Peering when the `peering_sync_level value` is `LocalNotInSync`.

Fixes #1023

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

azure_rm_virtualnetworkpeering

##### ADDITIONAL INFORMATION

1.) Add a CIDR block of address space to an existing peered VNet.

2.) The peering is now "Out of sync".

3.) The peering where the sync_level is "LocalNotInSync" needs to be re-synched.

```paste below
{
    "changed": false,
    "failed": false
}

{
    "changed": true,
    "failed": false,
    "peering_sync_level": "FullyInSync"
}
```

### CODE REVIEW

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 1]
```diff
@@ -98,6 +98,11 @@
     type: str
     sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/virtualNetworks/myVirtualN
              etwork/virtualNetworkPeerings/myPeering"
+peering_sync_level:
+    description:
+        - The Sync Level of the Peering
+    type: str
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py:104]:
```suggestion
    type: str
    returned: always
```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 2]
```diff
@@ -281,6 +289,15 @@ def exec_module(self, **kwargs):
 
             response = self.create_or_update_vnet_peering()
             self.results['id'] = response['id']
+            to_be_synced = self.check_sync(response)
+
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: remove whitespace
```suggestion

```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 3]
```diff
@@ -281,6 +289,15 @@ def exec_module(self, **kwargs):
...
             self.results['id'] = response['id']
+            to_be_synced = self.check_sync(response)
+        
+        if to_be_synced:
+            self.results['changed'] = True
+
+            if self.check_mode:
+                return self.results
+            sync_response = self.sync_vnet_peering()
+            self.results['peering_sync_level']  = sync_response['peering_sync_level']
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]:
```suggestion
            self.results['peering_sync_level'] = sync_response['peering_sync_level']
```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 4]
```diff
@@ -305,7 +322,12 @@ def format_vnet_id(self, vnet):
         else:
             self.fail("remote_virtual_network could be a valid resource id, dict of name and resource_group, name of virtual network in same resource group.")
         return remote_vnet_id
-
+
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: blank line contains whitespace
```suggestion

```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 5]
```diff
@@ -305,7 +322,12 @@ def format_vnet_id(self, vnet):
         else:
             self.fail("remote_virtual_network could be a valid resource id, dict of name and resource_group, name of virtual network in same resource group.")
         return remote_vnet_id
-
+    
+    def check_sync(self, exisiting_vnet_peering):
+        if exisiting_vnet_peering['peering_sync_level'] == 'LocalNotInSync':
+            return True
+        return False
+
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: blank line contains whitespace
```suggestion

```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 6]
```diff
@@ -330,6 +352,41 @@ def get_vnet(self, resource_group, vnet_name):
...
+                                     self.virtual_network['resource_group'])
+        peering = self.network_models.VirtualNetworkPeering(
+            id=vnet_id,
+            name=self.name,
+            remote_virtual_network=self.network_models.SubResource(id=self.remote_virtual_network),
+            allow_virtual_network_access=self.allow_virtual_network_access,
+            allow_gateway_transit=self.allow_gateway_transit,
+            allow_forwarded_traffic=self.allow_forwarded_traffic,
+            use_remote_gateways=self.use_remote_gateways
+            )
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: closing bracket does not match indentation of opening bracket's line
```suggestion
        )
```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 7]
```diff
@@ -330,6 +352,41 @@ def get_vnet(self, resource_group, vnet_name):
...
+                                                                                           self.virtual_network['name'],
+                                                                                           self.name,
+                                                                                           peering,
+                                                                                           sync_remote_address_space=True)
+            if isinstance(response, LROPoller):
+                response = self.get_poller_result(response)
+            return vnetpeering_to_dict(response)
+        except Exception as exc:
+            self.fail("Error creating Azure Virtual Network Peering: {0}.".format(exc.message))
+
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: blank line contains whitespace
```suggestion

```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 8]
```diff
@@ -350,7 +407,8 @@ def create_or_update_vnet_peering(self):
             allow_virtual_network_access=self.allow_virtual_network_access,
             allow_gateway_transit=self.allow_gateway_transit,
             allow_forwarded_traffic=self.allow_forwarded_traffic,
-            use_remote_gateways=self.use_remote_gateways)
+            use_remote_gateways=self.use_remote_gateways
+            )
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: closing bracket does not match indentation of opening bracket's line
```suggestion
        )
```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 9]
```diff
@@ -392,6 +450,7 @@ def get_vnet_peering(self):
                                                                         self.virtual_network['name'],
                                                                         self.name)
             self.log("Response : {0}".format(response))
+
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: blank line contains whitespace
```suggestion

```

**Review by Fred-sun — COMMENTED** [review]
sanity errors need to be improved

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 1 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py:104]: Added `returned: always`

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 3 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Fixed extra whitespace.

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 5 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Removed whitespace.

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 4 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Removed whitespace

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 6 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Removed whitespace

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 8 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Removed whitespace

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 9 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Removed whitespace

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 10]
```diff
@@ -330,6 +353,41 @@ def get_vnet(self, resource_group, vnet_name):
...
+                                                                                           self.virtual_network['name'],
+                                                                                           self.name,
+                                                                                           peering,
+                                                                                           sync_remote_address_space=True)
+            if isinstance(response, LROPoller):
+                response = self.get_poller_result(response)
+            return vnetpeering_to_dict(response)
+        except Exception as exc:
+            self.fail("Error creating Azure Virtual Network Peering: {0}.".format(exc.message))
+
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: trailing-whitespace: Trailing whitespace
```suggestion

```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 11]
```diff
@@ -330,6 +353,41 @@ def get_vnet(self, resource_group, vnet_name):
...
+                                     self.virtual_network['resource_group'])
+        peering = self.network_models.VirtualNetworkPeering(
+            id=vnet_id,
+            name=self.name,
+            remote_virtual_network=self.network_models.SubResource(id=self.remote_virtual_network),
+            allow_virtual_network_access=self.allow_virtual_network_access,
+            allow_gateway_transit=self.allow_gateway_transit,
+            allow_forwarded_traffic=self.allow_forwarded_traffic,
+            use_remote_gateways=self.use_remote_gateways
+            )
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: closing bracket does not match indentation of opening bracket's line
```suggestion
        )
```

File: plugins/modules/azure_rm_virtualnetworkpeering.py [hunk 12]
```diff
@@ -350,7 +408,8 @@ def create_or_update_vnet_peering(self):
             allow_virtual_network_access=self.allow_virtual_network_access,
             allow_gateway_transit=self.allow_gateway_transit,
             allow_forwarded_traffic=self.allow_forwarded_traffic,
-            use_remote_gateways=self.use_remote_gateways)
+            use_remote_gateways=self.use_remote_gateways
+            )
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_virtualnetworkpeering.py]: closing bracket does not match indentation of opening bracket's line
```suggestion
        )
```

**Review by Fred-sun — COMMENTED** [review]
small change request!

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 7 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Removed whitespace.

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 10 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Removed whitespace

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 11 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Fixed indentation.

File: plugins/modules/azure_rm_virtualnetworkpeering.py (hunk 12 above)
Comment (kmj251) [reply plugins/modules/azure_rm_virtualnetworkpeering.py]: Fixed indentation.

**Review by xuzhang3 — APPROVED** [review]

**Comment by xuzhang3** [comment]
@kmj251   LGTM

**Pull request merged**